lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -676,6 +686,14 @@ def write(chunk)
 
     private
 
+    def compute_trace(trace)
+      if @autoformat_stackdriver_trace &&
```

Comment:
The point here is that the important bit is the early return, not the clause. A reader, seeing `return trace unless [...]` will think: "Ah, this function will return one of two values — an untransformed one unless some condition holds true, and a transformed one otherwise". Then they can dig into the condition. In the code you're proposing, the logic goes the other way — the reader first has to get through a large condition to figure out how the value is transformed, and then look at the end of the function to see that it's returned.

Incidentally, even if we agree that your current structure is more readable, I think it's better to say `return "projects/#{@project_id}/traces/#{trace}"` instead of assigning it back to a parameter `trace` only to return it in the end.